test/functional/helper/AccessTokenRequester.groovy

```diff
@@ -16,7 +16,14 @@ class AccessTokenRequester {
     }
 
     static String getAccessToken(Map params) {
-        def response = requestAccessToken(params)
+        params = params.clone() // Don't mess with the original Map
+        String clientId = params.remove('client_id')
+        String clientSecret = params.remove('client_secret') ?: ''
+
+        def response = clientId != null ?
```

Comment:
My thoughts on this were that a missing `client_id` would result in a botched Authorization header, leading to a failed authentication, which is different from not providing any client credentials. 

After introducing the `NullRememberMeServices` I had 13 failed tests of this sort:

    junit.framework.AssertionFailedError: Assertion failed: 

    assert response.data.error_description == errorDescription
           |        |    |                 |  |
           |        |    Bad credentials   |  Full authentication is required to access this resource
           |        |                      false
           |        [error:unauthorized, error_description:Bad credentials]
           groovyx.net.http.HttpResponseDecorator@7df2e0f6